src/hooks/useFeedbackDisclosure.ts

```diff
@@ -0,0 +1,66 @@
+import { useDisclosure } from "@chakra-ui/react"
+import { useLocation } from "react-router-dom"
+
+import { LOCAL_STORAGE_KEYS } from "constants/localStorage"
+
+import { useLoginContext } from "contexts/LoginContext"
+
+import { isEditPageUrl, isSpecialPagesUrl } from "utils/pages"
+
+import { useLocalStorage } from "./useLocalStorage"
+
+type LastSeenFeedbackTime = number
+type UserId = string
+
+type FeedbackStorageMappings = Record<UserId, LastSeenFeedbackTime>
+
+const NPS_SURVEY_INTERVAL = 1814400000 as const // 3 weeks - 3 * 7 * 24 * 60 * 60 * 1000
```

Comment:
re: multiplication directly - sure, no worries will do that! did the `as const` to get it to infer exact type but maybe not worth.

re: 3 wks - actually we just need it to be the survey interval. i think i'll change the name + update time (1 wk) for that. the reason why is because we show it on a monthly basis for the duration of a week. if we set to a month, a person could do on the sunday of the prev month, and when we toggle it on again, only has the sunday to do it. hence, the time interval should be duration of survey.

re: constant - not incliend to. this constant isn't meant to be exported cos it's only used in 1 place for now and i want to hide it. if we got more use case for this, then i'll change its location + naming

Reply:
Sounds good, thanks for the explanation.

Reply:
@seaerchin wait, but if the time interval is 1 week, a person can do it every week in the same month, right?
As long as the it's been more than a week since he/she did it last time. 
Is this the intended product requirement?

Reply:
The env var will be off

Reply:
Only if duration + env var - the env var means that ppl need to do survey and the duration means whether they need to do the survey in this current week

Reply:
Got it @seaerchin. Yeah, that makes sense. Thanks for the clarification.